### PR TITLE
chore: rename sidechain-main-cli to pc-contracts-cli

### PR DESCRIPTION
**Pull request by tgunnoe**
This coorelates with the renaming in
https://github.com/input-output-hk/partner-chains-smart-contracts/pull/113 please do not merge until that PR is in master, and the flake input is updated properly in this PR.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [ ] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: .github/workflows/sidechain-main-cli-binary-host.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Build sidechain-main-cli and Push to EKS binary-host (optional)
+name: Build pc-contacts-cli and Push to EKS binary-host (optional)
```
Comment (LGLO) [inline .github/workflows/sidechain-main-cli-binary-host.yml:1]: typo

File: changelog.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ This changelog is based on [Keep A Changelog](https://keepachangelog.com/en/1.1.
 * polkadot-sdk dependency updated to partnerchains-stable2407 (stable2407 is v1.15.0 in the previous scheme)
 * changed the inner type of `McBlockHash` from Vec to an array
 * ETCM-7985 - bumped `partner-chains-smart-contracts` version and updated the `partner-chains-cli`
-to match. Now `partner-chains-cli` passes a network parameter to `sidechain-main-cli` where necessary.
+to match. Now `partner-chains-cli` passes a network parameter to `pc-contracts-cli` where necessary.
```
Comment (LGLO) [inline changelog.md:33]: I disagree with modification of past changes. I think, that instead this current change should be added on top of it.

Comment (tgunnoe) [reply changelog.md]: thanks, it was an overlook

File: partner-chains-cli/readme.md [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 A set of CLI style wizards to help getting started with Partner Chains.
 
-Should be packaged together with **partner-chains-node** executable, and **trustless-sidechain CLI**.
+Should be packaged together with **partner-chains-node** executable, and **partner-chain-smart-contracts CLI**.
```
Comment (LGLO) [inline partner-chains-cli/readme.md:5]: partner-chains-smart-contracts or pc-contracts-cli ?

**Review by LGLO — APPROVED** [review]
The PR should have a link to JIRA ticket that documents this requirement.

**Pull request merged**